### ISSUE TITLE
Update tests for scripts/papers/maxout

### ISSUE THREAD

**Issue by lamblin** [lisa-lab/pylearn2#1524]
There is already a test in pylearn2/models/tests/test_maxout.py, but it should be updated to:
- test the cuda-convnet layers if the GPU is recent enough,
- be automatically synchronized with the yaml files in scripts/papers/maxout if possible,
- check that even if the batch size does not divide the size of the training set, all examples are used

If possible, check if we can find a way of detecting if the size of the padding is not the right one.

See #569 for a more general ticket.
